src/pl/senla/hotel/application/di/DIApplication.java

```diff
@@ -0,0 +1,104 @@
+package pl.senla.hotel.application.di;
+
+import pl.senla.hotel.entity.Guest;
+import pl.senla.hotel.entity.Order;
+import pl.senla.hotel.entity.Hotel;
+import pl.senla.hotel.entity.facilities.HotelFacility;
+import pl.senla.hotel.entity.services.HotelService;
+import pl.senla.hotel.ie.serialization.Processor;
+import pl.senla.hotel.ie.serialization.ProcessorSerializable;
+import pl.senla.hotel.storage.*;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+
+public class DIApplication {
+
+    private static DIApplication application;
+    private DIApplication() {}
+    public static synchronized DIApplication getApplication() {
+        if (application == null) {
+            application = new DIApplication();
+        }
+        return application;
+    }
+
+    public static void run() throws InvocationTargetException,
+            IllegalAccessException, InstantiationException, NoSuchMethodException {
+        DIContext context = DIContext.getContext();
+        context.createDIContainer();
+        context.injectValuesFromDIContainer();
+        loadHotelData(context);
```

Comment:
loadHotelData, saveHotelData methods shouldn't be here, because it's infrastructural classes and they shouldn't know anything about your business operations or something like that, you have two options: 1) hardest one you can create a new annotation like springs post construct to call it when bean is configured to load your data from files 2) easier you can move it to repo layer to default constructor to load data right after bean creation

Reply:
ok. I'll create new Controller, Service and repo layers for entity Hotel and operations Load and Save data during Start/Exit application.

Reply:
I've got one more idea:
move calling **"loadingHoteldata()"** to StartMainMenu
![image](https://github.com/dudarenko-dmitry/senla_training/assets/102149674/670bae03-b59a-447f-b56e-638289a77b18)

and invoke **processor.saveHotelData()** in ExecutorMainMenu while the user choose "EXIT":
![image](https://github.com/dudarenko-dmitry/senla_training/assets/102149674/147f1258-d8f9-4cb0-96a8-97b5676d5629)